exporter/datadogexporter/integrationtest/integration_test.go

```diff
@@ -0,0 +1,280 @@
+	apmstatsRec := &testutil.HTTPRequestRecorderWithChan{Pattern: testutil.APMStatsEndpoint, ReqChan: make(chan []byte)}
+	tracesRec := &testutil.HTTPRequestRecorderWithChan{Pattern: testutil.TraceEndpoint, ReqChan: make(chan []byte)}
+	server := testutil.DatadogServerMock(apmstatsRec.HandlerFunc, tracesRec.HandlerFunc)
+	defer server.Close()
+
+	// 2. Start in-process collector
+	factories := getIntegrationTestComponents(t)
+	app, confFilePath := getIntegrationTestCollector(t, server.URL, factories)
+	go func() {
+		assert.NoError(t, app.Run(context.Background()))
+	}()
+	defer app.Shutdown()
+	defer os.Remove(confFilePath)
+	waitForReadiness(app)
+
+	// 3. Generate and send traces
+	sendTraces(t)
+
+	// 4. Validate traces and APM stats from the mock server
+	var spans []*pb.Span
+	tracesBytes := <-tracesRec.ReqChan
+	gz := getGzipReader(t, tracesBytes)
+	slurp, err := io.ReadAll(gz)
+	require.NoError(t, err)
+	var traces pb.AgentPayload
+	require.NoError(t, proto.Unmarshal(slurp, &traces))
+	for _, tps := range traces.TracerPayloads {
+		for _, chunks := range tps.Chunks {
+			spans = append(spans, chunks.Spans...)
+			for _, span := range chunks.Spans {
+				assert.Equal(t, span.Meta["_dd.stats_computed"], "true")
+			}
+		}
+	}
+
+	var stats []*pb.ClientGroupedStats
+	apmstatsBytes := <-apmstatsRec.ReqChan
+	gz = getGzipReader(t, apmstatsBytes)
+	var spl pb.StatsPayload
+	require.NoError(t, msgp.Decode(gz, &spl))
+	for _, csps := range spl.Stats {
+		for _, csbs := range csps.Stats {
+			stats = append(stats, csbs.Stats...)
+			for _, stat := range csbs.Stats {
+				assert.True(t, strings.HasPrefix(stat.Resource, "TestSpan"))
+				assert.Equal(t, stat.Hits, uint64(1))
+				assert.Equal(t, stat.TopLevelHits, uint64(1))
+			}
+		}
+	}
+
+	// 5 sampled spans + APM stats on 10 spans are sent to datadog exporter
+	assert.Len(t, spans, 5)
+	assert.Len(t, stats, 10)
+}
+
+func getIntegrationTestComponents(t *testing.T) otelcol.Factories {
+	var (
+		factories otelcol.Factories
+		err       error
+	)
+	factories.Receivers, err = receiver.MakeFactoryMap(
+		[]receiver.Factory{
+			otlpreceiver.NewFactory(),
+		}...,
+	)
+	require.NoError(t, err)
+	factories.Processors, err = processor.MakeFactoryMap(
+		[]processor.Factory{
+			batchprocessor.NewFactory(),
+			tailsamplingprocessor.NewFactory(),
+		}...,
+	)
+	require.NoError(t, err)
+	factories.Connectors, err = connector.MakeFactoryMap(
+		[]connector.Factory{
+			datadogconnector.NewFactory(),
+		}...,
+	)
+	require.NoError(t, err)
+	factories.Exporters, err = exporter.MakeFactoryMap(
+		[]exporter.Factory{
+			datadogexporter.NewFactory(),
+			loggingexporter.NewFactory(),
+		}...,
+	)
+	require.NoError(t, err)
+	return factories
+}
+
+func getIntegrationTestCollector(t *testing.T, url string, factories otelcol.Factories) (*otelcol.Collector, string) {
+	cfg := fmt.Sprintf(`
```

Comment:
nit: move config to separate file?

Reply:
This config needs to be here because I need to inject the (random) mock server URL as the datadog endpoint. I tried using and setting env vars `${env: URL}` but they seem not working with conf provider.

Any suggestions?

Reply:
:thinking: that's interesting, the `env` provider is loaded here https://github.com/open-telemetry/opentelemetry-collector/blob/d42d7e80974b3ba9fd1e235065638fe6a4d5455e/otelcol/otelcoltest/config.go#L24

Reply:
The error I got:
> failed to resolve config: cannot resolve the configuration: scheme "env" is not supported for uri "env:ENDPOINT"

This is where it came from https://github.com/open-telemetry/opentelemetry-collector/blob/2511161c7d8f94300c775be69c58537a5feba181/confmap/resolver.go#L203

I will open an issue in core and we can continue the discussion there.

Reply:
https://github.com/open-telemetry/opentelemetry-collector/issues/8757